sonic-chassisd/scripts/chassisd

```diff
@@ -167,6 +168,59 @@ class ModuleConfigUpdater(logger.Logger):
             self.log_info("Changing module {} to admin {} state".format(key, 'DOWN' if admin_state == MODULE_ADMIN_DOWN else 'UP'))
             try_get(self.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
 
+#
+# SmartSwitch Module Config Updater ========================================================
+#
+
+
+class SmartSwitchModuleConfigUpdater(logger.Logger):
+
+    def __init__(self, log_identifier, chassis):
+        """
+        Constructor for SmartSwitchModuleConfigUpdater
+        :param chassis: Object representing a platform chassis
+        """
+        super(SmartSwitchModuleConfigUpdater, self).__init__(log_identifier)
+
+        self.chassis = chassis
+        # The threading.Lock is required for concurrent setting of admin_state
+        # on multiple DPUs for faster bring-up and shutdown.
+        self.lock = threading.Lock()
+
+    def deinit(self):
+        """
+        Destructor of SmartSwitchModuleConfigUpdater
+        :return:
+        """
+
+    def module_config_update(self, key, admin_state):
```

Comment:
@rameshraghupathy could you please add a quick summary of function here and also remaining functions for future clarity?